### PR TITLE
Adds scaling for Kubernetes deployments through 'az ml service scale …

### PR DESCRIPTION
**Pull request by Chris113113**
…realtime -n <name> -z <num_replicas>'

---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] The PR has modified HISTORY.rst with an appropriate description of the change (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

### Command Guidelines

- [ ] Each command and parameter has a meaningful description.
- [ ] Each new command has a test.

(see [Authoring Command Modules](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules))

### CODE REVIEW

File: src/command_modules/azure-cli-ml/azure/cli/command_modules/ml/service/realtime.py [hunk 1]
```diff
@@ -249,14 +249,29 @@ def realtime_service_run_local(service_name, input_data, verbose):
 
 
 # TODO - not yet used
-def realtime_service_scale(context, args):
+def realtime_service_scale(service_name, num_replicas, context=cli_context):
```
Comment (scobb) [inline src/command_modules/azure-cli-ml/azure/cli/command_modules/ml/service/realtime.py]: nit, remove the TODO above :)

File: src/command_modules/azure-cli-ml/azure/cli/command_modules/ml/_params.py [hunk 2]
```diff
@@ -57,6 +57,8 @@
                       help='The data to use for calling the web service.', required=False)
 register_cli_argument('ml service create realtime', 'num_replicas', options_list=('-z',),
                       default=1, required=False, help='Number of replicas for a Kubernetes service.')
+register_cli_argument('ml service scale realtime', 'num_replicas', options_list=('-z',),
```
Comment (scobb) [inline src/command_modules/azure-cli-ml/azure/cli/command_modules/ml/_params.py]: nit: reduce this to "ml service" level and only have one register_cli_argument. This argument will only apply to functions that have a "num_replicas" arguments, so it's okay to go with broader scope.

File: src/command_modules/azure-cli-ml/azure/cli/command_modules/ml/service/realtime.py [hunk 3]
```diff
@@ -249,14 +249,29 @@ def realtime_service_run_local(service_name, input_data, verbose):
...
     if context.in_local_mode():
         print("Error: Scaling is not supported in local mode.")
         print("To scale a cluster based service, switch to cluster mode first: az ml env cluster")
         return
 
+    elif context.env_is_k8s:
+        try:
+            num_replicas = int(num_replicas)
+        except ValueError:
+            print("The -z option must be a non-negative integer.")
```
Comment (scobb) [inline src/command_modules/azure-cli-ml/azure/cli/command_modules/ml/service/realtime.py]: Doesn't actually validate the value is non-negative.
As discussed in chat, let's limit to 1-17.

File: src/command_modules/azure-cli-ml/azure/cli/command_modules/ml/_k8s_util.py [hunk 4]
```diff
@@ -299,6 +299,16 @@ def delete_replica_set(self, deployment_name):
             print("Unable to delete replica set for deployment {}. {} {}".format(deployment_name, exc, exc.output))
             raise
 
+    def scale_deployment(self, service_name, num_replicas):
+        try:
+            print("Scaling web service {} to {} pods".format(service_name, num_replicas))
+            deployment_name = service_name + '-deployment'
+            num_replicas = int(num_replicas)
+            subprocess.check_call(['kubectl', 'scale', 'deployment', deployment_name,
+                                   '--replicas={}'.format(num_replicas)])
```
Comment (scobb) [inline src/command_modules/azure-cli-ml/azure/cli/command_modules/ml/_k8s_util.py:308]: as discussed in chat, add a print statement to indicate that increasing pods may cause list/view to show as unhealthy.

**Review by scobb — CHANGES_REQUESTED** [review]

File: src/command_modules/azure-cli-ml/azure/cli/command_modules/ml/_k8s_util.py (hunk 4 above)
Comment (Chris113113) [reply src/command_modules/azure-cli-ml/azure/cli/command_modules/ml/_k8s_util.py:308]: Leaving as is per chat. It is required for 'scale' but optional for 'create'

**Review by scobb — APPROVED** [review]

**Pull request merged**